Added quote feature

davidecotten opened the pull request:
Likely to be abused
New command !quoth @mention saves messages to quote pickle file
New command !inspire creates an image from a random quote and image

davidecotten committed:
Added quote feature
Likely to be abused
New command !quoth @mention saves messages to quote pickle file
New command !inspire creates an image from a random quote and image

davidecotten committed:
Added super_waifu commands to !inspire

Added !viewquotes and !deletequote
Added checks for previously added quotes

aceat64 commented on bot.py:
```diff
@@ -109,6 +113,64 @@ def get_members_by_role(role):
+        with open(os.path.join(sys.path[0], 'quotes.dat'), 'rb') as fp:
+            return pickle.load(fp)
+    except FileNotFoundError:
+        quotes = []
```
Why not just have this be `return []`?

davidecotten replied:
Good question. Will be fixed.

aceat64 commented on bot.py:
```diff
@@ -109,6 +113,64 @@ def get_members_by_role(role):
+def create_quote_image(quote, name):
+    text = "\"{}\"".format(quote)            
+    name = "- {}".format(name)
+    file = "images/{}.jpg".format(random.randint(1,50))
```
Instead of blindly assuming images 1-5.jpg exist, how about we get a list of images in the folder and then randomly select one? That way we can add/remove images as needed easily.

davidecotten replied:
Fixed with `file = random.choice(os.listdir(os.path.join(sys.path[0], 'images')))`

aceat64 commented on bot.py:
```diff
@@ -251,7 +313,117 @@ def get_members_by_role(role):
+                await client.send_message(message.channel, reply_msg)
+                await asyncio.sleep(3)
+                reply_msg = ""
+                
```
There should be a return here, outside of the for loop.

davidecotten replied:
Yup.

aceat64 commented on bot.py:
```diff
@@ -109,6 +113,64 @@ def get_members_by_role(role):
+    draw.text((x,y),name,font=font, align='right', fill='white')
+
+    timestamp = datetime.datetime.now().strftime('%Y%m%d%H%M%S')
+    out_file = "images/{}.jpg".format(timestamp)
```
Lets create use a tmp folder for these generated images. We should also do something about cleaning up files after they've been uploaded.

davidecotten replied:
Moved generated images to new tmp folder. They were already being deleted using os.remove().

davidecotten committed:
Made some aceat64 requested changes
Simplified and added return statements
Changed image selection to random choice of files in 'images' dir
Created 'tmp' dir
